### DISCUSSION TITLE
firmware update

### DISCUSSION THREAD

**Discussion by thecaver99** [erichelgeson/BlueSCSI-v1#199] [General]
hi all,

if this in in the wrong section then i apiologies before hand

i have a 1.1a but it doesn't have the -usb firmware, i have tried to update via the st-link v2 but it will only allow me to update to BlueSCSI-v1.1-20220917-STM32F1.bin which is not the usb version.
i have flashed BlueSCSI-v1.1-20220917-STM32F1-USB.bin but the bluescsi doesn't flash only have red led lit.
if i go back to the none usb it flashes as it should.
can anyone help get the usb version on please.

i have read the github pages for flashing, i may have to replace the bluepill as no longer being recognized.
if flashing from scratch how would one flash the usb firmware and usb boot loader.

**Comment by erichelgeson** [comment]
If you use the script located here it will walk you through first flashing the usb bootloader and then flashing the latest firmware - https://github.com/erichelgeson/BlueSCSI-Updater

>  i have flashed BlueSCSI-v1.1-20220917-STM32F1-USB.bin but the bluescsi doesn't flash only have red led lit.

Correct, you need the usb bootloader flashed before - you can not flash the -USB firmware with an STLink

> i may have to replace the bluepill as no longer being recognized.

You can always recover a bluepill if you have an STLink. Use STM32 ST-LINK Utility and erase it a few times, then start again with the update script, following the directions closely.
